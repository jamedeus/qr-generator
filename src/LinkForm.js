import React from 'react';
import Form from 'react-bootstrap/Form';
import FloatingLabel from 'react-bootstrap/FloatingLabel';
import Button from 'react-bootstrap/Button';


function LinkForm({ onSubmit }) {
    return (
        <Form onSubmit={onSubmit}>
            <FloatingLabel label="URL" className="mb-3">
                <Form.Control type="text" name="url" placeholder="URL" />
            </FloatingLabel>

            <FloatingLabel label="Text" className="mb-3">
                <Form.Control type="text" name="text" placeholder="Optional" />
            </FloatingLabel>

            <div className="d-flex">
                <Button variant="primary" type="submit" className="mx-auto">
                    Generate
                </Button>
            </div>
        </Form>
    )
}


export default LinkForm;
