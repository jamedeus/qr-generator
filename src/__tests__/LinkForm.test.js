import { render } from '@testing-library/react';
import LinkForm from '../LinkForm';

describe('LinkForm', () => {
    it('matches snapshot', () => {
        const component = render(
            <LinkForm />
        );
        expect(component).toMatchSnapshot();
    });

    it('requires URL field, but not text field', async () => {
        const { getByPlaceholderText } = render(
            <LinkForm />
        );

        // Get reference to each field
        const urlField = getByPlaceholderText('URL');
        const textField = getByPlaceholderText('Optional');

        // Empty text field should be valid (optional)
        expect(textField.validity.valid).toBeTruthy();

        // URL field with default text (https://) should fail pattern match
        expect(urlField.validity.valid).toBeFalsy();
        expect(urlField.validity.patternMismatch).toBeTruthy();

        // Add arbitrary text, should fail pattern match
        urlField.value = "https://foobar";
        expect(urlField.validity.valid).toBeFalsy();
        expect(urlField.validity.patternMismatch).toBeTruthy();

        // Add valid URLs, should be valid with http, https, querystring, etc
        urlField.value = "https://foo.bar";
        expect(urlField.validity.valid).toBeTruthy();
        urlField.value = "http://foo.bar";
        expect(urlField.validity.valid).toBeTruthy();
        urlField.value = "https://foo.bar?param=value";
        expect(urlField.validity.valid).toBeTruthy();
    });
});
