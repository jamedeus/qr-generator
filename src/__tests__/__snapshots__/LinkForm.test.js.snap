// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkForm matches snapshot 1`] = `
<form
  className=""
  noValidate={true}
  onSubmit={[MockFunction]}
>
  <div
    className="mb-3 form-floating"
  >
    <input
      className="form-control"
      defaultValue="https://"
      name="url"
      pattern="http(s?):\\/\\/([a-zA-Z0-9\\-_]+\\.)*[a-zA-Z0-9\\-_]+\\.[a-zA-Z]{2,}([\\/\\?].*)?"
      placeholder="URL"
      required={true}
      type="text"
    />
    <label>
      URL
    </label>
  </div>
  <div
    className="mb-3 form-floating"
  >
    <input
      className="form-control"
      name="text"
      placeholder="Optional"
      type="text"
    />
    <label>
      Text
    </label>
  </div>
  <div
    className="d-flex"
  >
    <button
      className="mx-auto btn btn-primary"
      disabled={false}
      type="submit"
    >
      Generate
    </button>
  </div>
</form>
`;
